### DISCUSSION TITLE
Image Snapshots

### DISCUSSION THREAD

**Discussion by Metamogul** [utmapp/UTM#5087] [Ideas]
QEMU offers the possibility to create [file system snapshots ](https://kashyapc.fedorapeople.org/virt/lc-2012/snapshots-handout.html)of qcow2 files. This is useful if you for example want to test infrastructure provisioning with VMs instead of actual production metal, if you're researching malware, developing OS images, ... and frequently need to restore earlier system states. In such a use case a snapshot feature could safe a lot of disk space and disk usage compared to copying a VM('s disk). The config UI for a VM's drive also has a lot of empty space where a list of snapshots for restoring could be displayed.

Does a disk snapshot feature exist yet and I didn't find it? If not: Are there any known reasons why such a feature hasn't be implemented yet, like technical issues with snapshots via `qemu-img`, lack of working hours, low priority, ...? Do you think such a feature would be useful or useless?

**Comment by Metamogul** [comment]
Because I need that functionality myself (and because I'm not familiar with the architecture and intricacies of UTM) if created a proof-of-concept project: https://github.com/Metamogul/UTM-Snapshot-Manager. Let me know what you think!

**Comment by codecod3r** [comment]
Did you ever find a solution for this?

**Reply by Metamogul** [reply]
You mean a solution to the lack of snapshot functionality? I created the above mentioned tool as temporary solution until UTM will eventually support snapshots. You can easily clone and build the project if you need snapshots for VMs in UTM.
